src/blockchain/ethereum.rs

```diff
@@ -128,6 +129,36 @@ impl Ethereum {
                 ErrorKind::NodeError,
                 "No block found".to_string(),
             )),
+        });
+
+        // Get all events for that block from the node and add them to the block struct.
+        let web3_clone = self.web3.clone();
+        blocks.and_then(move |mut block| {
+            let block_number: u64 = block.number.into();
+            let block_number = BlockNumber::from(block_number);
+
+            // Filter for all logs of the current block.
+            let filter_builder = FilterBuilder::default();
+            let log_filter = filter_builder
```

Comment:
For every block, we are fetching logs(event). I was wondering if we have a separate worker for events. That can fetch the events in batches rather for every block.

whats your thoughts?  I may be missing some bigger picture here

Reply:
Ahh! One of the reasons to scan each block instead of batches can be that as a validator I want to respond quickly to an event

Reply:
I am not sure. Both could be valid approaches. I only did it this way, because I thought this is how platform does it 😅
Generally speaking, I think events can be very relevant, for example when another validator reports a block or proposes a meta-block. But maybe for those cases we should have an events-filter in addition to reading all events? 🤔
What do you think @deepesh-kn / @sarvesh-ost ?

Reply:
We decided to poll every block for now.